### PR TITLE
Dynamic Content filters fixed

### PR DESCRIPTION
**Pull request by annamunk**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Yes
| New feature/enhancement? (use the a.x branch)      | No
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes
| Related user documentation PR URL      | --
| Related developer documentation PR URL | --
| Issue(s) addressed                     | #11202

#### Description:
An error occurred when sending emails with dynamic content if the field contained in the filter belonged to a company and no company was assigned to the contact (company was null). PR solves this problem by skipping further checking the company field filter when the contact's company is missing.

#### Steps to test this PR:
Same as in "How can we reproduce this issue?" in #11202 but the campaign should be sending emails without errors.

### CODE REVIEW

**Comment by escopecz** [comment]
I run the unit tests without the fix and both are passing. Can you please build the test so it is failing before the code change and passing after?

**Review by volha-pivavarchyk — APPROVED** [review]
Works as described. The issue has been fixed.

branch 4.4
image

current branch:
image

But the new tests works without errors on two branches

**Review by jos0405 — APPROVED** [review]
This works perfectly.
The send worked even when the country didn't exist in the company profile.

image

**Comment by annamunk** [comment]
@volha-pivavarchyk @jos0405 @escopecz thank you for your reviews! I corrected the test so that it would not pass without applying the pull request changes.

File: app/bundles/EmailBundle/Tests/EventListener/MatchFilterForLeadTraitTest.php [hunk 1]
```diff
@@ -207,6 +207,35 @@ public function dataForInNotInOperatorFilter(): iterable
             false,
         ];
     }
+
+    public function testMatchFilterForLeadTraitEmptyCompany(): void
+    {
+        error_reporting(E_ALL);
```
Comment (escopecz) [inline app/bundles/EmailBundle/Tests/EventListener/MatchFilterForLeadTraitTest.php]: Is this necessary? It will affect other tests as well.

Comment (annamunk) [reply app/bundles/EmailBundle/Tests/EventListener/MatchFilterForLeadTraitTest.php]: This is not necessary, it was removed in the last commit.

**Review by escopecz — APPROVED** [review]
Thanks!

**Pull request merged**